packages/@n8n/db/src/repositories/execution.repository.ts

```diff
@@ -418,9 +418,20 @@ export class ExecutionRepository extends Repository<ExecutionEntity> {
 	async setRunning(executionId: string) {
 		const startedAt = new Date();
 
-		await this.update({ id: executionId }, { status: 'running', startedAt });
+		// Check if execution already has a startedAt timestamp
```

Comment:
I would like us to avoid making an extra query just to check this. It unnecessary extra load and could also cause a race condition (theoretical, this shouldn't be called twice anywhere). We can do this e.g. by coalescing the startedAt timestamp:

```sql
UPDATE execution_entity
SET
  startedAt = COALESCE(startedAt, :startedAt),
  status = 'running'
WHERE id = :id
```

or by ignoring it if it's already running

```sql
UPDATE execution_entity
SET
  startedAt = :startedAt,
  status = 'running'
WHERE id = :id AND status <> 'running'
```